packages/core/src/components/Search/Search.tsx

```diff
@@ -1,124 +1,127 @@
 import cx from "classnames";
-import React, { forwardRef } from "react";
-import TextField from "../TextField/TextField";
+import React, { forwardRef, useCallback, useRef } from "react";
 import useMergeRef from "../../hooks/useMergeRef";
-import { SearchDefaultIconNames, SearchType } from "./SearchConstants";
-import CloseIcon from "../Icon/Icons/components/CloseSmall";
+import CloseSmallIcon from "../Icon/Icons/components/CloseSmall";
 import SearchIcon from "../Icon/Icons/components/Search";
-import { NOOP } from "../../utils/function-utils";
-import { SubIcon, VibeComponentProps, VibeComponent, withStaticProps } from "../../types";
-import { TextFieldTextType } from "../TextField/TextFieldConstants";
-import { BASE_SIZES } from "../../constants";
 import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
 import styles from "./Search.module.scss";
+import BaseInput from "../BaseInput/BaseInput";
+import useDebounceEvent from "../../hooks/useDebounceEvent";
+import IconButton from "../IconButton/IconButton";
+import Icon from "../Icon/Icon";
+import { SearchComponent } from "./Search.types";
+import Loader from "../Loader/Loader";
 
-export interface SearchProps extends VibeComponentProps {
-  secondaryIconName?: SubIcon;
-  iconName?: SubIcon;
-  onChange?: (value: string) => void;
-  autoFocus?: boolean;
-  value?: string;
-  placeholder?: string;
-  disabled?: boolean;
-  debounceRate?: number;
-  onBlur?: (event: React.FocusEvent) => void;
-  onFocus?: (event: React.FocusEvent) => void;
-  wrapperClassName?: string;
-  setRef?: () => void;
-  autoComplete?: string;
-  /* BASE_SIZES is exposed on the component itself */
-  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
-  /* TYPES is exposed on the component itself */
-  type?: SearchType;
-  validation?:
-    | {
-        status: "error" | "success";
-        text: string;
-      }
-    | { text: string };
-  inputAriaLabel?: string;
-  searchResultsContainerId?: string;
-  activeDescendant?: string;
-  /*  Icon names labels for a11y */
-  iconNames?: {
-    layout: string;
-    primary: string;
-    secondary: string;
-  };
-  /** shows loading animation */
-  loading?: boolean;
-}
-
-const Search: VibeComponent<SearchProps, unknown> & {
-  sizes?: typeof BASE_SIZES;
-  types?: typeof SearchType;
-} = forwardRef<unknown, SearchProps>(
+const Search: SearchComponent = forwardRef(
   (
     {
-      secondaryIconName = CloseIcon,
-      iconName = SearchIcon,
-      onChange = NOOP,
-      autoFocus = false,
-      value = "",
-      placeholder = "",
-      disabled = false,
-      debounceRate = 200,
-      onBlur = NOOP,
-      onFocus = NOOP,
-      wrapperClassName = "",
-      setRef = NOOP,
+      searchIconName = SearchIcon,
+      clearIconName = CloseSmallIcon,
+      clearIconLabel = "Clear",
```

Comment:
in TextField the icon is not on the left, here you can have both icons simultaneously.
you have a good point since design also asked if we can avoid it
the thing is that we have cases where people changed the search icon and clear icon

and not sure about the single prop? I need to be able to pass label (as people would probably want to translate the label)

Reply:
I mean `clearIcon={<Icon icon={Clear} aria-label="clear" />}` and not sure if possible but we can use the SubIcon type but making the aria-label required in this case.
Plus if designers don't want it and we don't want to support it we can decide here to avoid it

Reply:
It should be an `IconButton`, though
and it can add complexity
in my best scenario it is removed, but it might be used by current consumers, making it require codemod and to remove ability

Reply:
I don't think codemod should prevent us from doing anything
You can use a render prop (e.g. `renderClearIcon` and provide props like we did in some other component. Just think about a case where users will want to add a `onClearIconClick` to trigger analytics for example, you'll have to add a prop for it